lib/src/async_import_cache.dart

```diff
@@ -153,18 +153,30 @@ final class AsyncImportCache {
           "in the browser.";
     }
 
-    if (baseImporter != null && url.scheme == '') {
-      var relativeResult = await putIfAbsentAsync(
-          _relativeCanonicalizeCache,
-          (
-            url,
-            forImport: forImport,
-            baseImporter: baseImporter,
-            baseUrl: baseUrl
-          ),
-          () => _canonicalize(baseImporter, baseUrl?.resolveUri(url) ?? url,
-              baseUrl, forImport));
-      if (relativeResult != null) return relativeResult;
+    if (url.scheme == '') {
+      if (baseImporter != null) {
+        var resolved = baseUrl?.resolveUri(url) ?? url;
+        var relativeResult = await putIfAbsentAsync(
+            _relativeCanonicalizeCache,
+            (
+              resolved,
```

Comment:
This cache key need to be `resolved`, despite that in some cases this is no longer "relative".

Because it's possible that the "second" importer is actually the same as baseImporter (which is used in embedded-host to emulate legacy API). We need to make sure baseImporter's cache key contains the actual attempted url, rather than original url, so that the original url can be attempted one more time in the "second" importer.